web/app.fluidity.money/app/queries/useVolumeTx.ts

```diff
@@ -23,113 +100,173 @@ export type VolumeTxsResponse = {
-              // is after timestamp
-              new Date(t.blockTimestamp) > new Date(iso8601Timestamp)
-            // sort descending by block timestamp
-          )
-          .sort(
-            (a, b) =>
-              new Date(b.blockTimestamp).getTime() -
-              new Date(a.blockTimestamp).getTime()
-          )
-          .map((t) => ({
-            sender: { address: t.fromAddress.format() },
-            receiver: { address: t.toAddress.format() },
-            block: {
-              timestamp: { unixtime: new Date(t.blockTimestamp).getTime() },
-            },
-            amount: t.value.toString(),
-            currency: { symbol: fluidAssets[t.address.checksum] },
-          })),
-      },
-    },
-  };
-  return filteredTransactions;
+      const filteredTransactions: VolumeTxsResponse = {
+        data: {
+          ethereum: {
+            transfers: transfers
+              .filter(
+                (t) =>
+                  // is a fluid token
+                  Object.keys(fluidAssets).includes(t.address.format()) &&
+                  // address is sender or receiver
+                  (t.fromAddress.equals(address) ||
+                    t.toAddress.equals(address)) &&
+                  // is after timestamp
+                  new Date(t.blockTimestamp) > new Date(iso8601Timestamp)
+                // sort descending by block timestamp
+              )
+              .sort(
+                (a, b) =>
+                  new Date(b.blockTimestamp).getTime() -
+                  new Date(a.blockTimestamp).getTime()
+              )
+              .map((t) => ({
+                sender: { address: t.fromAddress.format() },
+                receiver: { address: t.toAddress.format() },
+                block: {
+                  timestamp: { unixtime: new Date(t.blockTimestamp).getTime() },
+                },
+                amount: t.value.toString(),
+                currency: { symbol: fluidAssets[t.address.checksum] },
+              })),
+          },
+        },
+      };
+      return filteredTransactions;
+    }
+    case network === "ethereum":
+    case network === "solana": {
+      const variables = {
+        fluidAssets: Object.keys(fluidAssets),
+        address,
+        timestamp: iso8601Timestamp,
+      };
+      const url = "https://graphql.bitquery.io";
+      const body = {
+        variables,
+        query: queryByAddressTimestamp,
+      };
+      return jsonPost<VolumeTxsBodyByAddressTimestamp, VolumeTxsResponse>(
+        url,
+        body,
+        {
+          "X-API-KEY": process.env.FLU_BITQUERY_TOKEN ?? "",
+        }
+      );
+    }
+    default:
+      return {
+        errors: `Unsupported network ${network}`,
+      };
+  }
 };
 
 const useVolumeTxByTimestamp = async (
+  network: string,
   // address: token symbol
   fluidAssets: { [address: string]: string },
-  iso8601Timestamp: string
+  iso8601Timestamp: string,
+  useMoralis = true
```

Comment:
Add this as an options object, that way if we have to use other services it'll be as easy as spreading the option into the options argument.